RELEASE_NOTES.md

```diff
@@ -30,6 +30,9 @@ v1.0.0-alpha.xx
   module. These have been replaced by domain-specific traits and
   specifications defined in their own repositories, such as
   [OpenAssetIO-MediaCreation](https://github.com/OpenAssetIO/OpenAssetIO-MediaCreation).
+  Renamed remaining constants to have an `InfoKey_` prefix and migrated
```

Comment:
[fix] Can we add a `### Deprecated` section below breaking changes and note that the old prefixes will be removed in a future release?

Reply:
Added deprecated section in db9870c5